lib/flow_runner/custom_blocks/whatsapp_catalog.ex

```diff
@@ -0,0 +1,76 @@
+defmodule FlowRunner.CustomBlocks.WhatsAppCatalog do
+  @moduledoc """
+  A block to handle WhatsApp catalog messages.
+
+  WhatsApp catalog messages allow businesses to showcase their products
+  or services within the WhatsApp conversation. This block handles the
+  display of catalog information with optional footer text.
+
+  The `whatsapp_catalog` block compiles to this FLOIP block.
+  The `text` config parameter is required for the catalog body text.
+  The `footer` parameter is optional.
+  """
+
+  @behaviour FlowRunner.Spec.Block
+  use OpenTelemetryDecorator
+
+  @impl true
+  def validate_config!(%{
+        "catalog" =>
+          %{
+            "text" => text
+          } = params
+      }) do
+    optional_params = read_optional_params(params, [:footer])
+
+    catalog_config =
+      Map.merge(
+        %{
+          text: text
+        },
+        optional_params
+      )
+
+    %{catalog: catalog_config}
+  end
```

Comment:
Should we simplify this by not including the catalog key, as the block's type already suggests that it is a catalog?
```suggestion
  def validate_config!(%{"text" => text} = params) do
    optional_params = read_optional_params(params, [:footer])

    Map.merge(%{text: text}, optional_params)
  end
```